app/models/region.rb

```diff
@@ -53,6 +53,52 @@ def protected_areas_per_governance
     }
   end
 
+  def protected_areas_per_iucn_category
+    region_data = Hash.new { |hash, key| hash[key] = Hash.new }
+    processed_data = []
+    total_region_count = []
+
+    countries.each do |country|
+      country.protected_areas_per_iucn_category.each do |protected_area|
+        region_pa_category = region_data[protected_area["iucn_category_name"]]
```

Comment:
I think the problem lies here.
The `region_data` hash at this point is always empty, so it fails to retrieve the data inside `region_data[protected_area["iucn_category_name"]]` 
I'm not sure what your intent was here, but maybe you just need `protected_area["iucn_category_name"]` ?

Reply:
Thanks @Levia it was originally different, but it got changed to make it more concise. The purpose of that line:

```
region_pa_category = region_data[protected_area["iucn_category_name"]]
```

Was to make that protected_area's `iucn_category_name` the key inside the `region_data` hash. Which is then referenced to by the `region_pa_category` in subsequent lines.

I will try your suggestion!